http.c

```diff
@@ -289,17 +337,18 @@ static int read_timeout(int fd, int timeout_ms)
 }
 
 /* reads response, ignores fscking carriage returns */
-static int http_read_response(int fd, struct gbuf *buf, int timeout_ms)
+static int http_read_response(struct connection *conn, struct gbuf *buf, int timeout_ms)
 {
 	char prev = 0;
+	int fd = get_sockfd(conn);
```

Comment:
Adding `conn->has_read_timeout()` seems cleaner.

Also because SSL may have it's own BIO timeout settings that may not be applied on `fd` level.